storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -870,14 +870,14 @@ bool xbcloud_put(Object_store *store, const std::string &container,
   xtrabackup_tablespaces.00000000000000000000) is uploaded to cloud storage to
   determine successful xbcloud "put" operation. */
   std::string last_file_prefix = backup_name + "/xtrabackup_tablespaces";
+  auto last_file_size = last_file_prefix.size();
   bool file_found = false;
-
-  if (!object_list.empty()) {
-    auto cur_file = object_list.back();
-    auto last_file_size = last_file_prefix.size();
+  for (auto cur_file : object_list) {
```

Comment:
can we make this a bit more optimized and iterate `object_list` backward?
xtrabackup_tablespaces will always be on the last indexes of the array. The way we are doing right now will be a waste of CPU cycles knowing that what we are looking for is at the end.

Reply:
fixed.